packages/maskbook/src/plugins/NFT/UI/NFTInPost.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState } from 'react'
+import { useAsync } from 'react-use'
+import Services from '../../../extension/service'
+import MaskbookPluginWrapper from '../../MaskbookPluginWrapper'
+import type { NFTInPostProps, TokenDetails } from '../types'
+import { parseNftUrl } from '../utils'
+import NFTCardUI from './NFTCardUI'
+
+export default function NFTInPost(props: NFTInPostProps) {
+    const [tokenDetails, setTokenDetails] = useState<TokenDetails>({})
```

Comment:
Per documentation of https://github.com/streamich/react-use/blob/master/docs/useAsync.md, you can return the tokenDetails from the `useAsync`. It will be safer (in race condition situations)